pkg/rosa/nodepools.go

```diff
@@ -70,3 +74,73 @@ func (c *RosaClient) DeleteNodePool(clusterID string, nodePoolID string) error {
 	}
 	return nil
 }
+
+// CheckNodePoolExistingScheduledUpgrade checks and returns the current upgrade schedule for the nodePool if any.
+func (c *RosaClient) CheckNodePoolExistingScheduledUpgrade(clusterID string, nodePool *cmv1.NodePool) (*cmv1.NodePoolUpgradePolicy, error) {
+	upgradePolicies, err := c.getNodePoolUpgradePolicies(clusterID, nodePool.ID())
+	if err != nil {
+		return nil, err
+	}
+	for _, upgradePolicy := range upgradePolicies {
+		if upgradePolicy.UpgradeType() == cmv1.UpgradeTypeNodePool {
+			return upgradePolicy, nil
+		}
+	}
+	return nil, nil
+}
+
+// ScheduleNodePoolUpgrade schedules a new nodePool upgrade to the specified version at the specified time.
+func (c *RosaClient) ScheduleNodePoolUpgrade(clusterID string, nodePool *cmv1.NodePool, version string, nextRun time.Time) (*cmv1.NodePoolUpgradePolicy, error) {
+	// earliestNextRun is set to at least 5 min from now by the OCM API.
+	// we set it to 6 min here to account for latencty.
```

Comment:
Can you speak to what this means, what latency we are accounting for and why it is necessary?

Reply:
from the time we send the request to the time ocm process it.
so if we now() = 12:00:00 and we set nextRun = 12:05:00, by the time the request is received by OCM, now() could be 12:00:30 and ocm will decide the earliest nextRun is actually at 12:05:30 not 12:05:00